main/auth/external_login/facebook.inc.php

```diff
@@ -11,36 +11,61 @@
  * This files provides the facebook_connect()  and facebook_get_url functions
  * Please edit the facebook.conf.php file to adapt it to your fb application parameter
  * */
+
 require_once dirname(__FILE__) . '/../../inc/global.inc.php';
 require_once dirname(__FILE__) . '/facebook.init.php';
-require_once dirname(__FILE__) . '/facebook-php-sdk/src/facebook.php';
+
+require_once dirname(__FILE__) . '/facebook-php-sdk/autoload.php';
+
+use Facebook\FacebookSession;
+use Facebook\FacebookRedirectLoginHelper;
+use Facebook\FacebookRequest;
+use Facebook\FacebookResponse;
+use Facebook\FacebookSDKException;
+use Facebook\FacebookRequestException;
+use Facebook\FacebookAuthorizationException;
+use Facebook\GraphObject;
+use Facebook\Entities\AccessToken;
+use Facebook\HttpClients\FacebookCurlHttpClient;
+use Facebook\HttpClients\FacebookHttpable;
+
 require_once dirname(__FILE__) . '/functions.inc.php';
 
+FacebookSession::setDefaultApplication( $facebook_config['appId'],$facebook_config['secret']);
 /**
  * This function connect to facebook and retrieves the user info
  * If user does not exist in chamilo, it creates it and logs in
  * If user already exists, it updates his info
  * */
 function facebook_connect() {
-    global $facebook;
-    // See if there is a user from a cookie
-    $user = $facebook->getUser();
-    if ($user) {
-        try {
-            //Gets facebook user info
-            $fu = $facebook->api('/me');
-            $username = $fu['username'];
-            if (api_get_setting('login_is_email') == 'true' || empty($fu['username'])) {
-                $username = change_to_valid_chamilo_login($fu['email']);
-            }
-            //Checks if user already exists in chamilo
+
+    global $facebook_config;
+    global $helper; 
+    
+    try {
+	 $helper = new FacebookRedirectLoginHelper($facebook_config['return_url']);
+ 	 $session = $helper->getSessionFromRedirect();
+	// see if we have a session
+	if ( isset( $session ) ) {
+	   // graph api request for user data
+	    $request = new FacebookRequest( $session, 'GET', '/me' );
+	    $response = $request->execute();
+   	    // get response
+           $graphObject = $response->getGraphObject(); 
+           $username = change_to_valid_chamilo_login($graphObject->getProperty('email'));
+           $email = $graphObject->getProperty('email');
+	    $locale = $graphObject->getProperty('locale');
+	    $language = get_language($locale);
+	    if(!$language)$language='pl_PL';
```

Comment:
That's probably not a good idea (apart from the fact that spacing is wrong and that you should use brackets {}, the default language should be en_US